### PR TITLE
style: make `button` element inherit parent font globally

### LINKED ISSUE #668: "Hide All" text has a default font-family
## Bug description
Most text in the site has the DM sans font family. The "Hide All" text in signed tx has not font-family so is just using a default one

## Environment
 - Browser: any
 - Wallet: any
 - Chain: any

## Steps to reproduce
 1. Go to any safe with tx
 2. Open the tx details

Expected result
The text should have DM sans as font-family

## Screenshots
image

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves

Resolves #668

## How this PR fixes it
Sets a global style for `button` in order to keep the agent stylesheet from overwriting the font property

## How to test it
1. Go to the Tx Signers in the history list.
2. The "Hide All" button shall have `DM Sans,sans-serif` font-family

## Screenshots
Screenshot 2022-09-28 at 16 43 32

### CODE REVIEW

File: src/styles/theme.ts [hunk 1]
```diff
@@ -162,6 +162,13 @@ const initTheme = (darkMode: boolean) => {
           sizeLarge: { fontSize: '16px' },
         },
       },
+      MuiLink: {
+        styleOverrides: {
+          button: ({ theme }) => ({
+            fontFamily: theme.typography.fontFamily,
```
Comment (iamacook) [inline src/styles/theme.ts:168]: Can we not globally apply this to the entire theme?

Comment (DiogoSoaress) [reply src/styles/theme.ts:168]: I think the component MuiLink overrides it

Comment (usame-algan) [reply src/styles/theme.ts:168]: In the [docs](https://mui.com/material-ui/react-link/#accessibility) they also apply `variant="body2"` to the Link and it overrides the native browser stylesheet thats causing the issue with font-family.

Comment (mikhailxyz) [reply src/styles/theme.ts:168]: Why do we have to override the font family for this component in particular while we have the global font family set?

https://github.com/safe-global/web-core/blob/main/src/styles/theme.ts#L68-L69

Wouldn't it be a bug in the material UI if it doesn't work just for this component?

Comment (iamacook) [reply src/styles/theme.ts:168]: > overrides the native browser stylesheet thats causing the issue with font-family

The thing is, [we're MUI's `CSSBaseline`](https://github.com/safe-global/web-core/blob/main/src/pages/_app.tsx#L101).

Comment (iamacook) [reply src/styles/theme.ts:168]: [We need to override the `CssBaseline` as well](https://mui.com/material-ui/customization/typography/#self-hosted-fonts).

Comment (DiogoSoaress) [reply src/styles/theme.ts]: Even having the self-hosted fonts in the `MuiCssBaseline` override, the `button` font-family has to be overwritten it seems. Tried it in Chrome and Firefox.

**Comment by DiogoSoaress** [comment]
I've moved the font imports to the theme file as per the [documentation](https://mui.com/material-ui/customization/typography/#self-hosted-fonts)

File: src/styles/theme.ts [hunk 2]
```diff
@@ -109,6 +109,23 @@ const initTheme = (darkMode: boolean) => {
       },
     },
     components: {
+      MuiCssBaseline: {
+        styleOverrides: `
+          @font-face {
```
Comment (katspaugh) [inline src/styles/theme.ts:114]: Why does it have to be in the MUI theme and not in the global CSS? What's the difference?

Comment (DiogoSoaress) [reply src/styles/theme.ts:114]: I don't think there is a difference. They state
> In order to globally define Raleway as a font face, the [CssBaseline](https://mui.com/material-ui/react-css-baseline/) component can be used (or any other CSS solution of your choice).

Comment (DiogoSoaress) [reply src/styles/theme.ts:114]: Should I revert it? Also find it more readable in `globals.css`

Comment (iamacook) [reply src/styles/theme.ts]: If the font is defined in the theme, it can be used by others, e.g. when we export it in SRC.

Comment (DiogoSoaress) [reply src/styles/theme.ts]: Oh yes. Didn't think of that.

Comment (katspaugh) [reply src/styles/theme.ts]: Yes, no unnecessary changes is our motto.

**Review by katspaugh — APPROVED** [review]
👍

**Pull request merged**